### ISSUE TITLE
Best value for E/M step

### ISSUE THREAD

**Issue by gray-sunflower** [3205914485/FLiD#2]
Hi, could you provide the hyperparameters about the epochs to warm up or train in E/M step to reproduce the results in your paper. Moreover, I tried running the code but it give the error no value for graph_df.last_ts. I write a line of code to get the last timestamp of each unique source node. Could you providde the preprocess code or check if whether the line is correct.

![Image](https://github.com/user-attachments/assets/7ea60de1-4072-4749-ad25-1c75ed468e8f)

 Thanks you!

**Comment by 3205914485** [comment]
We have updated the repository to provide the preprocess scripts.
As for the hyperparameters, most can be found in the paper, or just set as the default in the args.
The warmup epochs were all set to 100.

**Comment by gray-sunflower** [comment]
Hi, Thanks for updating the code. However, i saw the preprocess code that produce the `last_u_ts` and `last_i_ts` for wikipedia but in the dataloader `last_u_ts` and `last_i_ts` are used for double_way_dataset which doesn't include wikipedia. For wikipedia or reddit, do i have to set label_times = last_u_ts. Moreover, what is the best value for `num_epochs_e_step` and `num_epochs_m_step` in the experiments. Thanks you!

**Comment by 3205914485** [comment]
For one_way_dataset, it's correct to set label_times to last_u_ts. The preprocess scripts we provide are for the double_way_dataset, which is relatively hard to process.
`num_epochs_e_step` and `num_epochs_m_step` were set to 50.

**Comment by gray-sunflower** [comment]
Hi, sorry for bothering you again. I want to ask about the code of E-step and M-step. In the paper, the E-step optimize only the decoder and the M-step optimize the encoder. However, i saw in the code that the function `e_step` optimize both encoder and decoder while the `m_step` i only saw the training of the decoder. Please feel free to correct me if i miss understood something. Many Thanks

**Comment by 3205914485** [comment]
We use different optimizers to control which part to train, the `e_step` only trains the encoder while set the `args.decoder == 1` for PTCL; While the `e_step` trians the encoder-decoder together as setting the `args.decoder == 2` for PTCL-2D.
Thanks for figuring out the problem of "E" "M" name mix, we forgot to fix it as we first name the procedure of training the encoder as `E-step`, we will fix it immediately!
